manifests/mta.pp

```diff
@@ -27,10 +27,20 @@
 #
 class postfix::mta (
   Pattern[/^\S+(?:,\s*\S+)*$/]               $mydestination = $postfix::mydestination,
-  Pattern[/^(?:\S+?(?:(?:,\s)|(?:\s))?)*$/]  $mynetworks    = $postfix::mynetworks,
+  Variant[Array[String], String]             $mynetworks    = $postfix::mynetworks,
   Pattern[/^\S+$/]                           $relayhost     = $postfix::relayhost,
 ) {
 
+  if (type($mynetworks) =~ Type[Array[String]]) {
+    $_mynetworks = join($mynetworks,',')
+  } else {
+    $_mynetworks = $mynetworks
+  }
+
+  if $_mynetworks !~ /^(?:\S+?(?:(?:,\s)|(?:\s))?)*$/ {
+    fail '$mynetworks don\'t match the pattern /^(?:\S+?(?:(?:,\s)|(?:\s))?)*$/'
```

Comment:
Why not keep this validation in the Data Type?

Reply:
I didn't believe it was possible to have a data type defined as Variant[Array[String], Pattern[/^(?:\S+?(?:(?:,\s)|(?:\s))?)*$/]]